### ISSUE TITLE
Format on save option

### ISSUE THREAD

**Issue by radogado** [patrickvuarnoz/nova-beautify#2]
Hi, thanks for the extension. This is a question, not an issue: how to make it format on save? Thanks.

**Comment by patrickvuarnoz** [comment]
Currently that feature is not supported. If it get's more "votes" I'll have a look into it.

**Comment by jblechinger** [comment]
This feature would be so nice!

**Comment by patrickvuarnoz** [comment]
Just released version 1.5 of the extension with format on save option. Please provide feedback.

**Comment by jblechinger** [comment]
Very cool 👍 
Thanks for your quick implementation! Its cool that you can choose where to use format on save.
Would be handy if you have the option to skip formating on save in some cases (with special shortcut?).

**Comment by gregwolanski** [comment]
Thank you, Patrick – it’s amazing! :)

**Comment by jblechinger** [comment]
And one thing i also have a problem with now is that i cant have trailing whitespaces at the end of files.
Before nova added this automatically on save. Now its gone because of beautify.

**Comment by patrickvuarnoz** [comment]
> And one thing i also have a problem with now is that i cant have trailing whitespaces at the end of files. Before nova added this automatically on save. Now its gone because of beautify.

This looks like a Nova issue. Format on save is invoked onWillSave() and I would expect that Nova should handle its own saving processing AFTER any extension did their work which obviously is not the case. I filed a bug report with Panic.

**Comment by problematic1** [comment]
When formatting on save, could it maintain "Code Folding" if used?

**Comment by patrickvuarnoz** [comment]
> When formatting on save, could it maintain "Code Folding" if used?

I don't think that an extension can handle this. The extension just reads text ranges and replaces them with the formatted text. There's no info about code folding. If Nova was smart enough to recognize that only the content but not the structure has changed, then code folding might be maintained.

**Comment by ZevEisenberg** [comment]
It looks like this may be supported now. Can this issue be closed?

<img width="998" height="977" alt="Image" src="https://github.com/user-attachments/assets/fdad6ef8-23b4-4a23-ad5a-5b685df546ff" />

**Comment by patrickvuarnoz** [comment]
@ZevEisenberg Sure. Format on save is supported since late 2022. Kept the ticket open to collect feedback.
